src/openssl/ssl/socket.cr

```diff
@@ -179,4 +179,16 @@ abstract class OpenSSL::SSL::Socket < IO
       String.new(host_name)
     end
   end
+
+  # Returns the cipher currently in use
+  def cipher : String
+    String.new(LibSSL.ssl_cipher_get_name(LibSSL.ssl_get_current_cipher(@ssl)))
+  end
+
+  # Returns the TLS version currently in use
+  def tls_version : String?
+    if version = LibSSL.ssl_get_version(@ssl)
```

Comment:
This function returns `"unknown"` when no version has been set, not `null` as stated [here](https://linux.die.net/man/3/ssl_get_version). Should this maybe be
```cr
if (version = LibSSL.ssl_get_version(@ssl)) == "unknown"`
```
?

Reply:
Ok, maybe a test case for that too, how did you test it?

Reply:
Ok, or just return "unknown" instead of being smart?

Reply:
I think when it can return `"unknown"` then that should become a `nil` but I wonder if it's even possible that this method returns `"unknown"`? I don't think it's possible to call this without a connection being established. But I'm not sure.

Reply:
No, this should simply return the value returned by `ssl_get_version`. If that's `"unknown"`, that's fine. But the semantics shouldn't change to `nil`.